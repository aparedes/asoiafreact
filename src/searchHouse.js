/* @flow */
import React, { Component } from 'react'
import { connect } from 'react-redux'

type PropsType = {

}
class SearchHouse extends Component {
  static displayName = 'SearchHouse'

  constructor(props: PropsType) {
    super(props)
    this.onChange = this.onChange.bind(this)
  }
  props: PropsType
  onChange: (evt: KeyboardEvent) => void

  onChange(evt: KeyboardEvent) {
    const input: HTMLInputElement = evt.target
    console.log(input.value)
  }
  render() {
    return (
      <input onChange={ this.onChange } placeholder={ 'Search house' }/>
    )
  }
}

const mapStateToProps = (state: Object): Object => ({

})

const mapDispatchToProps = (dispatch: (action: Object) => void): Object => ({

})

export default connect(mapStateToProps, mapDispatchToProps)(SearchHouse)
