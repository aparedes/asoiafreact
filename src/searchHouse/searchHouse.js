/* @flow */
import './searchHouse.css';
import React, { Component } from 'react';
import { connect } from 'react-redux';

type Props = {
  setSearch: (search: string) => void,
};

export class SearchHouse extends Component<Props> {
  static displayName = 'SearchHouse';

  onChange: (evt: KeyboardEvent) => void;

  constructor(props: Props) {
    super(props);
    this.onChange = this.onChange.bind(this);
  }

  onChange(evt: KeyboardEvent) {
    const input = evt.target;
    console.log(input.value);
    this.props.setSearch(input.value);
  }

  render() {
    return (
      <input
        className={'search'}
        onChange={this.onChange}
        placeholder={'Search house'}
      />
    );
  }
}

const mapStateToProps = (state: Object): Object => ({});

const mapDispatchToProps = (dispatch: (action: Object) => void): Object => ({
  setSearch: (search: string) => dispatch({ type: 'SET_SEARCH', search }),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchHouse);
