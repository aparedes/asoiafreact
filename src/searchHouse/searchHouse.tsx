import './searchHouse.css';
import React, { Component, ChangeEvent } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from '../store/constants/reduxTypes';
import { AppEnum } from '../store/reducers/app';

interface Props extends DTP {}
interface KeyEvt {
  target: { value: string };
}

export class SearchHouse extends Component<Props> {
  static displayName = 'SearchHouse';

  constructor(props: Props) {
    super(props);
    this.onChange = this.onChange.bind(this);
  }

  onChange(evt: ChangeEvent<HTMLInputElement>) {
    const input = evt.target;
    console.log(input.value);
    this.props.setSearch(input.value);
  }

  render() {
    return (
      <input
        className={'search'}
        onChange={this.onChange}
        placeholder={'Search house'}
      />
    );
  }
}

interface DTP {
  setSearch: (search: string) => void;
}
const mapDispatchToProps = (dispatch: Dispatch): DTP => ({
  setSearch: (search: string) => dispatch({ type: AppEnum.SET_SEARCH, search }),
});

export default connect(null, mapDispatchToProps)(SearchHouse);
