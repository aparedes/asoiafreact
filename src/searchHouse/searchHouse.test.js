import React from 'react';
import { SearchHouse } from './searchHouse';
import renderer from 'react-test-renderer';

test('Prints a list of houses', () => {
  const component = renderer.create(<SearchHouse />);
  let searchHouse = component.toJSON();
  expect(searchHouse).toMatchSnapshot();
});
