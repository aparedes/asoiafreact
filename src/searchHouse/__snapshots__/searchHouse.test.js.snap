// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prints a list of houses 1`] = `
<input
  className="search"
  onChange={[Function]}
  placeholder="Search house"
/>
`;
